content/refer-other-arguments-in-args.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: sip
+title: SIP-32 - Allow referring to other arguments in default parameters
+vote-status: pending
+permalink: /sips/:title.html
+redirect_from: /sips/pending/refer-other-arguments-in-args.html
+---
+
+**By: Pathikrit Bhowmick**
+
+## History
+
+| Date          | Version          |
+|---------------|------------------|
+| Jan 11th 2017 | Initial Draft    |
+| Jan 12th 2017 | Initial Feedback |
+| Jan 16th 2017 | Minor Changes    |
+
+## Introduction
+Currently there is no way to refer to other arguments in the default parameters list:
+
+Does not compile:
+```scala
+def substring(s: String, start: Int = 0, end: Int = s.length): String
+```
+
+The workaround to achieve this is by using a curried-function:
+```scala
+def substring(s: String, start: Int = 0)(end: Int = s.length): String
+```
+
+However, the above workaround is not always suitable in all situations since you may not want a curried function.
+
+The other more verbose alternative is by overloading:
+```scala
+def substring(s: String, start: Int): String
+  = substring(s, start = 0, end = s.length)
+def substring(s: String, start: Int = 0, end: Int): String
+```
+
+The above is quite verbose as it required 1 extra function definition per argument that refers other args.
+
+### Proposal
+Allow to refer to ***any*** parameters in the same (or left) curried parameter list:
+```scala
+def substring(s: String, start: Int = 0, end: Int = s.length)   // Legal
+def substring(start: Int = 0, end: Int = s.length, s: String)   // Legal !!!
```

Comment:
I don't think this should be legal. As mentioned at the end of the proposal, no major language allows this. And probably for a good reason: it would be very hard to specify in what *order* default values would be evaluated in this case.

I suggest to remove these references to the right from the proposal, as I don't think it is likely to be viable with them.

Reply:
I agree. This would be confusing.